Source/WindowsShellExtension/dllmain.cpp

```diff
@@ -0,0 +1,299 @@
+        data.resize(stringLengthInWchars);
+        if (data.compare(L"true") == 0)
+            return true;
+        if (data.compare(L"false") == 0)
+            return false;
+        throw std::runtime_error("Not a boolean.");
+    }
+}
+
+struct ExplorerCommandHandler : public winrt::implements<ExplorerCommandHandler, IExplorerCommand> {
+public:
+    // IExplorerCommand implementation:
+
+    IFACEMETHODIMP GetTitle(_In_opt_ IShellItemArray* items, _Outptr_ PWSTR* name) {
+        // Provide name for display in File Explorer context menu entry
+        UNREFERENCED_PARAMETER(items);
+        return SHStrDupW(menu_entry_title, name);
+    }
+
+    IFACEMETHODIMP GetIcon(_In_opt_ IShellItemArray* items, _Outptr_ PWSTR* icon) {
+        // Provide icon for display in File Explorer context menu entry
+        // Get path to application exe and use it as source for icon
+        UNREFERENCED_PARAMETER(items);
+        std::filesystem::path module_path{ wil::GetModuleFileNameW<std::wstring>(wil::GetModuleInstanceHandle()) };
+        module_path = module_path.remove_filename();
+        module_path /= exe_filename;
+        return SHStrDupW(module_path.c_str(), icon);
+    }
+
+    IFACEMETHODIMP GetToolTip(_In_opt_ IShellItemArray* items, _Outptr_ PWSTR* infoTip) {
+        UNREFERENCED_PARAMETER(items);
+        *infoTip = nullptr;
+        return E_NOTIMPL;
+    }
+
+    IFACEMETHODIMP GetCanonicalName(_Out_ GUID* guidCommandName) {
+        *guidCommandName = GUID_NULL;
+        return S_OK;
+    }
+
+    IFACEMETHODIMP GetState(_In_opt_ IShellItemArray* items, _In_ BOOL okToBeSlow, _Out_ EXPCMDSTATE* cmdState) {
+        // Provide state of File Explorer context menu entry
+        // Hide it if registry setting indicates that it should be disabled, else it is enabled
+        UNREFERENCED_PARAMETER(items);
+        UNREFERENCED_PARAMETER(okToBeSlow);
+        try {
+#ifdef MEDIAINFO_QT
+            if (!RegGetBool(HKEY_CURRENT_USER, L"Software\\MediaArea.net\\MediaInfo", L"shellExtension"))
+#else
+            if (!RegGetDword(HKEY_CURRENT_USER, L"Software\\MediaArea\\MediaInfo", L"ShellExtension"))
+#endif // MEDIAINFO_QT
+            {
+                *cmdState = ECS_HIDDEN;
+                return S_OK;
+            }
+        }
+        catch (...) {
+            // Error reading reg, default enabled
+        }
+        // Default enabled if not disabled
+        *cmdState = ECS_ENABLED;
+        return S_OK;
+    }
+
+    IFACEMETHODIMP GetFlags(_Out_ EXPCMDFLAGS* flags) {
+        *flags = ECF_DEFAULT;
+        return S_OK;
+    }
+
+    IFACEMETHODIMP EnumSubCommands(_Outptr_ IEnumExplorerCommand** enumCommands) {
+        *enumCommands = nullptr;
+        return E_NOTIMPL;
+    }
+
+    IFACEMETHODIMP Invoke(_In_opt_ IShellItemArray* items, _In_opt_ IBindCtx* bindCtx) {
+        // Process items passed by File Explorer when context menu entry is invoked
+        UNREFERENCED_PARAMETER(bindCtx);
+        if (items) {
```

Comment:
```suggestion
        if (!items) {
            return S_OK;
        }
```


And the other part 4 spaces smaller, instead, no need to go up to end of the function if no `items`.

Reply:
Microsoft's developer wrote it like that. Changed now.